python/setup.py

```diff
@@ -131,6 +131,7 @@
     "click >= 7.0",
     "colorama",
     "colorful",
+    "dataclasses; python_version < '3.7'",
```

Comment:
@richardliaw did we move this accidentally recently? Python 3.6 used to work (and we have dependend on dataclasses in ray client since last year).

Reply:
I don't recall; though I think we didn't have the python version setup correctly before. dataclasses will break on 3.7.